### PR TITLE
feat: Hedgebar

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Toolbars are boring. Why not have a Hedgehog walking around your website providing you with useful information?

## Changes

* Adds the Hedgehog mode as an option for an alternative Toolbar button
* Adds accessories for the Hedgehog - atm only working when opened in PostHog

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## Still todo
- [x] Some style things todo with the circles still visible on the hedgehog.
- [x] A new circle button for toggling the hedgehog

### CODE REVIEW

File: frontend/src/lib/components/HedgehogBuddy/HedgehogBuddy.tsx [hunk 1]
```diff
@@ -395,34 +449,66 @@ export function HedgehogBuddy({ onClose }: { onClose: () => void }): JSX.Element
...
-                                {capitalizeFirstLetter(x)}
-                            </LemonButton>
+                popoverOverlay || (
+                    <div className="HedgehogBuddyPopover p-2 max-w-140">
+                        <h3>Hi, I'm Max!</h3>
+                        <p>
+                            Don't mind me. I'm just here to keep you company.
+                            <br />
+                            You can move me around by clicking and dragging or control me with WASD / arrow keys. See if
+                            you can unlock all my accessories the more you use PostHog...
```
Comment (pauldambra) [inline frontend/src/lib/components/HedgehogBuddy/HedgehogBuddy.tsx:463]:
```suggestion
                            You can move me around by clicking and dragging or control me with WASD / arrow keys.
```

nit 😊

**Review by pauldambra — APPROVED** [review]
🚀

File: frontend/src/lib/components/HedgehogBuddy/HedgehogBuddy.tsx [hunk 2]
```diff
@@ -1,16 +1,28 @@
...
+} from './sprites/sprites'
+import { FlaggedFeature } from '../FlaggedFeature'
+import { FEATURE_FLAGS } from 'lib/constants'
+import { HedgehogBuddyAccessory } from './components/AccessoryButton'
+import './HedgehogBuddy.scss'
 
 const xFrames = SPRITE_SHEET_WIDTH / SPRITE_SIZE
 const boundaryPadding = 20
-const FPS = 24
+const FPS = 24 // 24
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/src/lib/components/HedgehogBuddy/HedgehogBuddy.tsx:25]: No idea :D

File: frontend/src/lib/components/HedgehogBuddy/hedgehogbuddyLogic.ts [hunk 3]
```diff
@@ -17,8 +20,38 @@ export const hedgehogbuddyLogic = kea<hedgehogbuddyLogicType>([
...
+    selectors({
+        availableAccessories: [
+            () => [],
+            () => {
+                // Until we figure out how to do this - all accessories are available
+                return Object.keys(standardAccessories)
+                // return Object.keys(standardAccessories).filter((x) => {
+                //     const key = `hedgehog-accessory-${x}`
+                //     return featureFlags[key]
+                // })
```
Comment (mariusandra) [inline frontend/src/lib/components/HedgehogBuddy/hedgehogbuddyLogic.ts:50]:
```suggestion
```

File: frontend/src/lib/components/HedgehogBuddy/sprites/sprites.tsx [hunk 4]
```diff
@@ -54,12 +89,12 @@ export const standardAnimations: { [key: string]: SpriteInfo } = {
...
-        frames: 9,
-        maxIteration: 3,
-        randomChance: 2,
-    },
+    // spin: {
+    //     img: hhSpin,
+    //     frames: 9,
+    //     maxIteration: 3,
+    //     randomChance: 2,
+    // },
```
Comment (mariusandra) [inline frontend/src/lib/components/HedgehogBuddy/sprites/sprites.tsx:97]: This is not code

**Review by mariusandra — APPROVED** [review]
Why not. Just some comments on committing in dead code, but 🚢 🦔

**Pull request merged**